yeat/tests/test_oxford.py

```diff
@@ -23,9 +23,8 @@ def test_oxford_nanopore_assemblers_dry_run(tmp_path):
 @pytest.mark.parametrize("algorithm", ["flye", "canu", "unicycler"])
 def test_oxford_nanopore_read_assemblers(algorithm, capsys, tmp_path):
     wd = str(tmp_path)
-    cores = str(get_core_count())
+    cores = get_core_count()
     config = write_config(algorithm, wd, "ont.cfg")
-    arglist = ["-o", wd, "-t", cores, config]
+    arglist = ["-o", wd, "-t", str(cores), config]
```

Comment:
Where is this `get_core_count()` function defined? Is there a way to limit or control this? If not, it could be problematic when running tests in CI or HPC environments.

Reply:
The `get_core_count()` is located in `yeat/tests/__init__.py`. I created this function to get the core count of a system because these tests are incredibly long. I thought the more cores we can throw at it, the faster we can run these assemblies. We can use the default `core=1` parameter (make our assembly tests very slow) or we can hard cap it to 4. Up for any suggestions!

> Is there a way to limit or control this?

As of right now, we are not limiting or controlling this. We can control it if we want by setting a cap or limit.

```
import multiprocessing

def get_core_count():
    return multiprocessing.cpu_count()
```

> If not, it could be problematic when running tests in CI or HPC environments.

Yes, I do agree with you here.

Reply:
Will create an issue thread on this!